analyses/TX_cd_2020/03_sim_TX_cd_2020.R

```diff
@@ -0,0 +1,445 @@
+
+z <- geomander::seam_geom(map$adj, map, admin = "cluster_edge", seam = c(0, 1))
+
+z <- z[z$cluster_edge == 1, ]
+
+border_idxs <- which(m3$row_id %in% z$row_id)
+
+constraints <- redist_constr(m3) %>%
+    add_constr_grp_hinge(
+        20,
+        cvap_hisp,
+        total_pop = cvap,
+        tgts_group = c(0.45)
+    ) %>%
+    add_constr_grp_hinge(
+        10,
+        cvap_black,
+        total_pop = cvap,
+        tgts_group = c(0.45)) %>%
+    add_constr_custom(strength = 10, function(plan, distr) {
+        ifelse(any(plan[border_idxs] == 0), 0, 1)
+    })
+
+map %>% st_drop_geometry() %>%
+    filter(county %in% clust3) %>%
+    group_by(cd_2020) %>%
+    summarise(hisp_prop = sum(cvap_hisp)/sum(cvap),
+        black_prop = sum(cvap_black)/sum(cvap))
+
+n_steps <- (sum(m3$pop)/attr(map, "pop_bounds")[2]) %>% floor()
+
+dallas_plans <- redist_smc(m3, counties = county,
+    nsims = nsims, n_steps = n_steps,
+    constraints = constraints)
+
+p <- redist.plot.plans(dallas_plans, draws = c(10, 20, 30, 50), m3)
+ggsave("data-raw/dallas.pdf")
+#############################################################
+## Combine Clusters
+
+houston_plans$dist_keep <- ifelse(houston_plans$district == 0, FALSE, TRUE)
+austin_plans$dist_keep <- ifelse(austin_plans$district == 0, FALSE, TRUE)
+dallas_plans$dist_keep <- ifelse(dallas_plans$district == 0, FALSE, TRUE)
+
+tx_plan_list <- list(list(map = m1, plans = houston_plans),
+    list(map = m2, plans = austin_plans),
+    list(map = m3, plans = dallas_plans))
+
+prep_mat <- prep_particles(map = map, map_plan_list = tx_plan_list,
+    uid = row_id, dist_keep = dist_keep, nsims = nsims)
+
+## Check contiguity
+test_vec <- sapply(1:ncol(prep_mat), function(i) {
+    cat(i, "\n")
+    z <- map %>%
+        mutate(ex_dist = ifelse(prep_mat[, i] == 0, 1, 0))
+
+    z <- geomander::check_contiguity(adj = z$adj, group = z$ex_dist)
+
+    length(unique(z$component[z$group == 1]))
+})
+
+table(test_vec)/nsims
+
+constraints <- redist_constr(map) %>%
+    add_constr_grp_hinge(
+        5,
+        cvap_hisp,
+        total_pop = cvap,
+        tgts_group = c(0.45)
+    ) %>%
+    add_constr_grp_hinge(
+        2,
+        cvap_black,
+        total_pop = cvap,
+        tgts_group = c(0.45))
+
+map <- set_pop_tol(map, 0.01)
+
+plans <- redist_smc(map, nsims = nsims,
+    counties = county, verbose = TRUE,
+    constraints = constraints, init_particles = prep_mat,
+    pop_temper = 0.01)
+
+plans <- plans %>% filter(draw != "cd_2020")
+plans <- plans %>% add_reference(ref_plan = map$cd_2020)
+
+cli_process_done()
+cli_process_start("Saving {.cls redist_plans} object")
+
+# Output the redist_map object. Do not edit this path.
+write_rds(plans, here("data-out/TX_2020/TX_cd_2020_plans.rds"), compress = "xz")
+cli_process_done()
+
+# Compute summary statistics -----
+cli_process_start("Computing summary statistics for {.pkg TX_cd_2020}")
+
+plans <- add_summary_stats_cvap(plans, map)
```

Comment:
to avoid duplicating `add_summary_stats()` code, I'm going to change this back to `add_summary_stats()` and then add the CVAP columns by hand.  I'll take care of it; just want to avoid the situation where we add/update summary stats in the main function and then forget to make the same change here, in FL, in CA, etc.